### DISCUSSION TITLE
How is lf shell expanding $fx?

### DISCUSSION THREAD

**Discussion by avidseeker** [gokcehan/lf#1603] [Q&A] [answered]
This line works as expected
```
cmd ls ${{
	ls $fx
}}
```
Now, launch a shell from lf: `$zsh`, then
```sh
$ ls $fx
```
This results in
```
ls: cannot access '/home/user/fileA'$'\n''/home/user/fileB': No such file or directory
```
What does lf use internally to expand $fx so that it is split based on $IFS? I would expected something like `ls $(echo $fx)`, but I'm still confused why isn't shell variable splitting doesn't apply here.

**Comment by gokcehan** [comment]
@avidseeker This behavior is implemented [here](https://github.com/gokcehan/lf/blob/d654f09885cc9838b175ccf110d1e01ad0970658/os.go#L137). If I'm not mistaken, when you start a new shell (e.g. `IFS="\n"; zsh`), it does not inherit `IFS`, and that's why you get a different behavior as opposed to directly running a command (e.g. `IFS="\n"; ls $fx`).

**Reply by avidseeker** [reply]
This is what I get after `$zsh` within `lf`.

```sh
$ echo -n $IFS | hexdump -C
00000000  20 09 0a 00                                       | ...|
```
Which is space, tab and new line. Even after launching `$zsh` and setting `IFS="\n"`, the ls command returns the same error. Not sure why $fx is not split.

**Reply by gokcehan** [reply, marked as answer]
@avidseeker I think maybe `zsh` does not split by default. We have a tip about it in our wiki page [here](https://github.com/gokcehan/lf/wiki/Tips#split-words-by-default-in-zsh)
